File: DataGenerator/domain/teacher.py


class Teacher:

    def __init__(self, username, name):
        self.__username = username
        self.__name = name

    @property
    def username(self):
        return self.__username

    @property
    def name(self):
        return self.__name

    def __str__(self):
        return "'" + self.__username + "'"

File: AcademicWebApp/src/main/java/com/example/AcademicWebApp/Models/Teacher.java
package com.example.AcademicWebApp.Models;


import lombok.*;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;

@Data
@Entity(name = "teacher")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Teacher {

    @Id
    @Column(name = "username")
    private String username;

    @Override
    public String toString() {
        return "Teacher{" +
                "username='" + username + '\'' +
                '}';
    }
}
